enhancements/custom-alert-configuration.md

```diff
@@ -0,0 +1,438 @@
+eviction or PDBs were never designed with such use cases in mind.
+KubeVirt use case could be explored in a KEP, but that is a different topic.
+
+Users that wish to use Live Migration feature then trigger this alert. From
+their perspective they are just using KubeVirt API as intended and this warning
+is unnecessary. The warning is a way to tell people that there are not enough
+replicas in their workload  and that the node drain might not work.
+For KubeVirt that is not the case - the intended number of pods per VM is 1
+and the eviction is handled in a custom way.
+
+For users with live migratable VMs, this alert is a constant occurrence.
+Apart from the alert being annoying, this diminishes the value of the alert,
+and it could be very easily overlooked when other workloads run into the PDB
+limit.
+
+### User Stories
+
+Components should be able to mark certain resources in the cluster that should
+not trigger specific alerts. However, this should only be allowed with the
+consent of the alert owner.
+
+
+* As an OpenShift component owner I want to be able to have a mechanism to opt
+  out of alerts, where it makes sense.
+
+* As a KubeVirt user I do not want to see alerts for PDBs that are guarding
+  virtual machines with `LiveMigrate` eviction strategy.
+
+### Goals
+
+- Introduce a generic CRD for alert exclusion that can be used by the
+  ValidatingAdmissionPolicy.
+- Create a ValidatingAdmissionPolicy and alert for
+  kube-controller-manager-operator that acknowledges PDB labels.
+
+### Non-Goals
+
+## Proposal
+
+### Workflow Description
+
+To solve this we need a mechanism for customizing alerts.
+We are proposing to introduce a
+`alerts.openshift.io/AlertName: excluded` label to allow exclusion of a set of
+specific objects from a specific alert. In the case of KubeVirt we would like
+to exclude PodDisruptionBudgets owned by KubeVirt from
+`PodDisruptionBudgetAtLimit` alert. So the following label would have to be
+placed on these PDBs:
+`alerts.openshift.io/PodDisruptionBudgetAtLimit: excluded`.
+The owner of the alert would have to make sure to exclude these objects in the
+alert query.
+
+Nevertheless, only allowed actors should have the power to set these labels to
+prevent misuse.
+
+In the case of the `PodDisruptionBudgetAtLimit` alert, excluding other users'
+PDBs from the alerting would have a detrimental effect on the cluster
+stability. Users could set this on any PDB and admins would have no reliable
+way of knowing whether something is blocking the node maintenance or an
+upgrade.
+
+We can use an alternative admission control based on [CEL][5] expressions that
+does not have the maintenance and reliability issues of an admission webhook.
+We can create [ValidatingAdmissionPolicy][6] for PDB objects to prevent a misuse
+of these labels. The validation expression would test for a presence of a
+specified label in the PDB object and would check for the Kubevirt user/SA
+in the request to decide if it is allowed.
```

Comment:
I don't see how using an admission webhook makes it impossible for cluster admins to add the same label ``alerts.openshift.io/AlertName: excluded`` alert to their PDBs.

Reply:
nothing will make it impossible for cluster admins to add this label, because they can block or disassemble any guards that folks try to put in place.  I think the bar here is either "we assume all PDB owners will play nice" or "we assume cluster-admins and a privileged subset of PDB owners will play nice, but are suspicious about PDB owners we don't know", and currently the enhancement is aiming for the latter.